R/deployApp.R

```diff
@@ -272,21 +266,15 @@ deployApp <- function(appDir = getwd(),
     application <- applicationForTarget(client, accountDetails, target, forceUpdate)
   })
 
-  if (isCloudServer(accountDetails$server) && !is.null(appVisibility)) {
-    # Shinyapps defaults to public visibility.
-    # Other values should be set before data is deployed.
-    currentVisibility <- application$deployment$properties$application.visibility
-    if (is.null(currentVisibility)) {
-      currentVisibility <- "public"
-    }
-
-    if (!identical(appVisibility, currentVisibility)) {
-      withStatus(paste0("Setting visibility to ", appVisibility), {
-        client$setApplicationProperty(application$id,
-                                   "application.visibility",
-                                   appVisibility)
-      })
-    }
+  # Change _visibility_ before uploading data
```

Comment:
Alternatively, hide the control logic, too:

```r
conditionallyChangeVisibility(accountDetails, application, appVisibility, quiet = quiet)
```

Reply:
Yeah, for sure, that's a bigger win.

Reply:
Hmmm, I tried it and I didn't like — you also need to pass `client` (so that's a whole grab bag of arguments), and I kind of like keeping all the big picture status updates in the main function.

Reply:
Good point; I missed `client` in that main function. Pity that the server / account don't hang off the client object...